pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotRealtimeTableResource.java

```diff
@@ -104,6 +110,70 @@ public Response pauseConsumption(
     }
   }
 
+  @POST
+  @Produces(MediaType.APPLICATION_JSON)
+  @Path("/tables/{tableName}/reload")
+  @ApiOperation(value = "Reloads the table across all the servers", notes = "This would reconstruct the table data"
+      + "manager in case of configuration changes. Example usage: trigger after converting the upsert mode"
+      + "from full to partial.")
+  public SuccessResponse reloadTable(
+      @ApiParam(value = "Name of the table", required = true) @PathParam("tableName") @Nullable
+      String tableName,
+      @ApiParam(value = "Whether to force the reload (if table config is not updated, reload will not be done)")
+      @QueryParam("force") boolean force,
+      @Context HttpHeaders httpHeaders, @Context Request request) {
+    String tableNameWithType = TableNameBuilder.REALTIME.tableNameWithType(tableName);
```

Comment:
This should apply to both offline and real-time table

Reply:
Good point, updated this API to have an optional tableTypeStr param.

If it's not passed - both the offline and realtime versions (whichever is available) of the table would be reloaded.